docs/quickstart/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,926 @@
+Typically, custom codes are required in only those areas.
+4- Add your upgradable config in params/precompile_config.go
+5- Add your precompile upgrade in params/config.go
+6- Add your solidity interface and test contract to contract-examples/contracts
+7- Write solidity tests for your precompile in contract-examples/test
+8- Create your genesis with your precompile enabled in tests/e2e/genesis/
+9- Create e2e test for your solidity test in tests/e2e/solidity/suites.go
+10- Run your e2e precompile Solidity tests with 'E2E=true ./scripts/run.sh'
+*/
+```
+
+### Step 1: Set Contract Address
+
+In `./precompile/hello_world.go`, we can see our precompile address is set to some default value. We can cut the address from the var declaration block and remove it from the precompile.
+
+![Singleton StatefulPrecompiledContract and signatures](./../../static/img/2022-09-01-22-46-00.png)
+
+We can paste it here in `./precompile/params.go`.
+
+```go
+ContractDeployerAllowListAddress = common.HexToAddress("0x0200000000000000000000000000000000000000")
+	ContractNativeMinterAddress      = common.HexToAddress("0x0200000000000000000000000000000000000001")
+	TxAllowListAddress               = common.HexToAddress("0x0200000000000000000000000000000000000002")
+	FeeConfigManagerAddress          = common.HexToAddress("0x0200000000000000000000000000000000000003")
+	HelloWorldAddress                = common.HexToAddress("0x0200000000000000000000000000000000000004")
+	// ADD YOUR PRECOMPILE HERE
+	// {YourPrecompile}Address       = common.HexToAddress("0x03000000000000000000000000000000000000??")
+```
+
+```go
+UsedAddresses = []common.Address{
+		ContractDeployerAllowListAddress,
+		ContractNativeMinterAddress,
+		TxAllowListAddress,
+		FeeConfigManagerAddress,
+		HelloWorldAddress,
+		// ADD YOUR PRECOMPILE HERE
+		// YourPrecompileAddress
+	}
+```
+
+Now when Subnet-EVM sees the `HelloWorldAddress` as input when executing [`CALL`](https://github.com/ava-labs/subnet-evm/blob/master/core/vm/evm.go#L222), [`STATICCALL`](https://github.com/ava-labs/subnet-evm/blob/master/core/vm/evm.go#L401), [`DELEGATECALL`](https://github.com/ava-labs/subnet-evm/blob/master/core/vm/evm.go#L362), [`CALLCODE`](https://github.com/ava-labs/subnet-evm/blob/master/core/vm/evm.go#L311), it can [run the precompile](https://github.com/ava-labs/subnet-evm/blob/master/core/vm/evm.go#L271-L272) if the precompile is enabled.
+
+### Step 2: Set Gas Costs
+
+Set up gas costs. In `precompile/params.go` we have `writeGasCostPerSlot` and `readGasCostPerSlot`. This is a good starting point for estimating gas costs.
+
+```go
+// Gas costs for stateful precompiles
+const (
+	writeGasCostPerSlot = 20_000
+	readGasCostPerSlot  = 5_000
+)
+```
+
+For example, we will be getting and setting our greeting with `sayHello()` and `setGreeting()` in one slot respectively so we can define the gas costs as follows.
+
+```go
+	SayHelloGasCost uint64    = 5000
+	SetGreetingGasGost uint64 = 20_000
+```
```

Comment:
Please specify that these two are in `hello_world.go`.

Reply:
Thanks